hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java

```diff
@@ -2339,9 +2340,16 @@ boolean snapshotCleanupSwitch(final boolean on, final boolean synchronous)
    * @param logQueryFilter filter to be used if provided (determines slow / large RPC logs)
    * @return online slowlog response list
    * @throws IOException if a remote or network exception occurs
+   * @deprecated since 2.4.0 and will be removed in 4.0.0.
+   *   Use {@link #getLogEntries(LogRequest, int)} instead.
    */
-  List<OnlineLogRecord> getSlowLogResponses(final Set<ServerName> serverNames,
-      final LogQueryFilter logQueryFilter) throws IOException;
+  default List<OnlineLogRecord> getSlowLogResponses(final Set<ServerName> serverNames,
```

Comment:
No 'limit' parameter here. Improvement should track the changes we are making to the generic API? Add a method and leave this one for backwards compat?

Reply:
This is deprecated API and by IA definitions, we can't change the signature of this without breaking minor release compatibility? That's why we made this default implementation to use new API internally.

Reply:
Yes, this is what I meant by "add a method and leave this one for backwards compat". Additional methods on an interface are binary compat per Java spec and our compat guidelines as long as existing method signatures are maintained. 

If the LogQueryFilter can impose a limit, then that would work too.